java/src/main/java/org/digma/intellij/plugin/idea/psi/java/MicronautFramework.java

```diff
@@ -0,0 +1,156 @@
+                if (controllerClass == null) {
+                    continue; // very unlikely
+                }
+                PsiAnnotation controllerAnnotation = controllerClass.getAnnotation(CONTROLLER_ANNOTATION_STR);
+                if (controllerAnnotation == null) {
+                    continue; // skip this method, since its class is not a controller
+                }
+                String endpointUriPrefix = JavaLanguageUtils.getPsiAnnotationAttributeValue(controllerAnnotation, "value");
+
+                String methodCodeObjectId = JavaLanguageUtils.createJavaMethodCodeObjectId(currPsiMethod);
+                String httpEndpointCodeObjectId = createHttpEndpointCodeObjectId(currPsiMethod, currAnnotation, endpointUriPrefix);
+                if (httpEndpointCodeObjectId == null) {
+                    continue; // skip this method, since endpoint value could not be determined
+                }
+
+                EndpointInfo endpointInfo = new EndpointInfo(httpEndpointCodeObjectId, methodCodeObjectId, documentInfo.getFileUri());
+                Log.log(LOGGER::debug, "Found endpoint info '{}' for method '{}'", endpointInfo.getId(), endpointInfo.getContainingMethodId());
+
+                MethodInfo methodInfo = documentInfo.getMethods().get(endpointInfo.getContainingMethodId());
+                //this method must exist in the document info
+                Objects.requireNonNull(methodInfo, "method info " + endpointInfo.getContainingMethodId() + " must exist in DocumentInfo for " + documentInfo.getFileUri());
+                methodInfo.addEndpoint(endpointInfo);
+            }
+        });
+    }
+
+    @Nullable
+    protected static String createHttpEndpointCodeObjectId(PsiMethod psiMethod, JavaAnnotation httpMethodAnnotation, String endpointUriPrefix) {
+        PsiAnnotation httpPsiAnnotation = psiMethod.getAnnotation(httpMethodAnnotation.getClassNameFqn());
+        List<JvmAnnotationAttribute> annotationAttributes = httpPsiAnnotation.getAttributes();
+
+        String endpointUriSuffix = "/";
+        for (JvmAnnotationAttribute curr : annotationAttributes) {
+            // taking the first attribute, either "value" or "uri" - that's how micronaut behave if both exists
+            if (VALUE_OR_URI.contains(curr.getAttributeName())) {
+                endpointUriSuffix = JavaLanguageUtils.getPsiAnnotationAttributeValue(httpPsiAnnotation, curr.getAttributeName());
+                if (endpointUriSuffix == null) {
+                    Log.log(LOGGER::debug, "cannot create http endpoint for method '{}' since could not extract attribute value for name '{}' from annotation '{}'",
+                            psiMethod.getName(), curr.getAttributeName(), httpMethodAnnotation.getClassNameFqn());
+                    return null; // unlikely
+                }
+                break; // found the first occurrence, and out
+            }
+            //TODO: handle AttributeName of uris (String[])
```

Comment:
is this TODO required for future implementation or maybe it's just forgotten?

Reply:
I think its irrelevant, I will take it out

Reply:
fine! thanks!